Client/Impl/Worker/JobWorker.cs

```diff
@@ -77,10 +86,95 @@ internal void Open()
         {
             isRunning = true;
             var cancellationToken = source.Token;
-            var jobWorkerSignal = new JobWorkerSignal();
 
-            StartPollerThread(jobWorkerSignal, cancellationToken);
-            StartHandlerThreads(jobWorkerSignal, cancellationToken);
+            activateJobsCommand = jobWorkerBuilder.Command;
+            maxJobsActive = jobWorkerBuilder.Command.Request.MaxJobsToActivate;
+
+            // Create options for blocks
+            var bufferOptions = new DataflowBlockOptions
+            {
+                CancellationToken = cancellationToken,
+                EnsureOrdered = false
+            };
+
+            var executionOptions = new ExecutionDataflowBlockOptions
+            {
+                MaxDegreeOfParallelism = jobWorkerBuilder.ThreadCount,
+                CancellationToken = cancellationToken,
+                EnsureOrdered = false
+            };
+
+            // Buffer for polled jobs
+            var input = new BufferBlock<IJob>(bufferOptions);
+
+            // Transformblock to process polled jobs
+            var transformer = new TransformBlock<IJob, IJob>(async activatedJob =>
+                {
+                    try
+                    {
+                        await jobHandler(jobClient, activatedJob);
```

Comment:
The job client is not thread safe, we need to create a new job client wrapper for each call

Reply:
I did not really understand the concept of the wrapper. What exactly is not  thread-safe?

Reply:
Got your point. ClientWasUsed is not thread safe.

Reply:
:+1: